php/class-wp-seo-social-settings.php

```diff
@@ -0,0 +1,708 @@
+								'name' => $key,
+								'content' => $og_img_src[0],
+							);
+						}
+					} else {
+						$tags[] = array(
+							'name' => $key,
+							'content' => $value,
+						);
+					}
+				}
+			}
+			$arbitrary_tags = array_merge(
+				$arbitrary_tags,
+				$tags
+			);
+			return $arbitrary_tags;
+		} );
+	}
+
+	/**
+	 * Additional sanitization for our new fields
+	 *
+	 * @param  array $out The options currently being saved.
+	 * @param  array $in  The options, raw.
+	 * @return array $out The options to save.
+	 */
+	public function sanitize( $out, $in ) {
+		$sanitize_as_text_field = $this->handle_as_text;
+		foreach ( $sanitize_as_text_field as $field ) {
+			$out[ $field ] = isset( $in[ $field ] ) && ( is_string( $in[ $field ] ) || is_integer( $in[ $field ] ) ) ? sanitize_text_field( $in[ $field ] ) : null;
+		}
+		return $out;
+	}
+
+	/**
+	 * Set class properties.
+	 */
+	public function set_properties() {
+		/**
+		 * Filter the post types that support per-entry social fields.
+		 *
+		 * @param array Associative array of post type keys and objects.
+		 */
+		$this->single_post_types = apply_filters(
+			'wp_seo_social_single_post_types',
+			wp_list_filter(
+				get_post_types( array( 'public' => true ), 'objects' ),
+				array( 'label' => false ),
+				'NOT'
+			)
+		);
+
+		/**
+		 * Filter the taxonomies that support social fields on term archive pages.
+		 *
+		 * @param  array Associative array of taxonomy keys and objects.
+		 */
+		$this->taxonomies = apply_filters(
+			'wp_seo_social_taxonomies',
+			wp_list_filter(
+				get_taxonomies(
+					array(
+						'public' => true,
+					),
+					'objects'
+				),
+				array(
+					'label' => false,
+				),
+				'NOT'
+			)
+		);
+
+		/**
+		 * Filter the post types that support social fields on their archive pages.
+		 *
+		 * @param array Associative array of post type keys and objects.
+		 */
+		$this->archived_post_types = apply_filters(
+			'wp_seo_social_archived_post_types',
+			wp_list_filter(
+				get_post_types(
+					array( 'has_archive' => true ),
+					'objects'
+				),
+				array( 'label' => false ),
+				'NOT'
+			)
+		);
+
+		/**
+		 * Filter the options to save by default.
+		 *
+		 * These are also the settings shown when the option does not exist,
+		 * such as when the the plugin is first activated.
+		 *
+		 * @param  array Associative array of setting names and values.
+		 */
+		$this->default_options = apply_filters(
+			'wp_seo_social_default_options',
+			array(
+			'post_types' => array_keys( $this->single_post_types ),
+				'taxonomies' => array_keys( $this->taxonomies ),
+			)
+		);
+
+		/**
+		 * Setup the options for easy access.
+		 */
+		$this->options = get_option(
+			'wp-seo',
+			$this->default_options
+		);
+	}
+
+	/**
+	 * Register the plugin settings.
+	 */
+	public function register_settings() {
+		$social_sections = array(
+			array(
+				'section'  => 'social_post_types',
+				'title'    => __( 'Social Post Types', 'wp-seo-social' ),
```

Comment:
I see you are using `wp-seo-social` and `wp-seo` as the text domains throughout the PR.  Should this only have a single text domain?